src/index.js

```diff
@@ -15,6 +21,21 @@ function randomUint() {
   return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
 }
 
+function untilDataChannelOpen(dataChannel) {
+  return new Promise(resolve => {
+    if (dataChannel.readyState === "open") {
+      resolve();
+    } else {
+      const resolver = () => {
+        dataChannel.removeEventListener("open", resolver);
+        resolve();
+      };
+
+      dataChannel.addEventListener("open", resolver);
```

Comment:
Add an event listener for "error" on the data channel and reject so that the `createPublisher` method doesn't hang indefinitely when erroring.